core/src/main/java/hudson/model/TaskListener.java

```diff
@@ -155,5 +155,5 @@ default PrintWriter fatalError(String format, Object... args) {
     /**
      * {@link TaskListener} that discards the output.
      */
-    TaskListener NULL = new StreamTaskListener(new NullStream());
+    TaskListener NULL = NullPrintStream::new;
```

Comment:
nit: maybe add `@FunctionalInterface` to the interface declaration for clarity (assuming that it would not impose any additional backwards compatibility constraints).